# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieService.h
# Commit: b2c86eba88a3
# Full Hash: b2c86eba88a348fb99ed75b0c1aed59848c0d451
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-09-17 17:41 +0000
# Regressor Bug: 1982261
# File Overlap Count: 1
# Overlapping Files: netwerk/cookie/CookieService.cpp
# Description:
#   Bug 1982261 - CookieServiceParent logs about invalid cookies instead of crashing,  a=dmeehan DONTBUILD
# ==============================================================================

diff -r b889a3daa95f -r b2c86eba88a3 netwerk/cookie/CookieService.h
--- a/netwerk/cookie/CookieService.h	Wed Sep 17 13:33:37 2025 +0000
+++ b/netwerk/cookie/CookieService.h	Wed Sep 17 17:41:13 2025 +0000
@@ -16,6 +16,7 @@
 #include "ThirdPartyCookieBlockingExceptions.h"
 
 #include "nsString.h"
+#include "nsICookieValidation.h"
 #include "nsIMemoryReporter.h"
 #include "mozilla/MemoryReporting.h"
 #include "mozilla/MozPromise.h"
@@ -100,11 +101,11 @@
                   const nsACString& aName, const nsACString& aPath,
                   bool aFromHttp, const nsID* aOperationID);
 
-  bool SetCookiesFromIPC(const nsACString& aBaseDomain,
-                         const OriginAttributes& aAttrs, nsIURI* aHostURI,
-                         bool aFromHttp, bool aIsThirdParty,
-                         const nsTArray<CookieStruct>& aCookies,
-                         dom::BrowsingContext* aBrowsingContext);
+  nsICookieValidation::ValidationError SetCookiesFromIPC(
+      const nsACString& aBaseDomain, const OriginAttributes& aAttrs,
+      nsIURI* aHostURI, bool aFromHttp, bool aIsThirdParty,
+      const nsTArray<CookieStruct>& aCookies,
+      dom::BrowsingContext* aBrowsingContext);
 
  protected:
   virtual ~CookieService();